addons/api/addon/handlers/client-daemon-handler.js

```diff
@@ -0,0 +1,67 @@
+import { pluralize } from 'ember-inflector';
+
+/**
+ * Not all types are yet supported by the client daemon so we'll
+ * just whitelist the ones we need to start with.
+ * @type {string[]}
+ */
+const supportedTypes = ['target', 'session'];
+
+/**
+ * Handler to sit in front of the API request layer
+ * so we can request from the daemon first
+ */
+const ClientDaemonHandler = {
+  async request(context, next) {
+    switch (context.request.op) {
+      case 'query': {
+        const { store, data } = context.request;
+        const { type, query } = data;
+        const isClientDaemonRunning = await this.ipc.invoke(
+          'isClientDaemonRunning'
+        );
+
+        if (!supportedTypes.includes(type) || !isClientDaemonRunning) {
```

Comment:
Should I just default back to our normal calls if the daemon isn't running for any reason? Or should we error out?

Reply:
That is a great question. I would imagine we want to default back to our normal calls. We wouldn't be able to do any search and filtering though right? Should we hide the bar if so?

Reply:
Yes exactly, we wouldn't be able to properly filter, it probably would make sense to hide the filters if that's the route we go down but we haven't talked too much about the failure case